File: C++/475. Heaters.cpp
class Solution {
public:
    int findRadius(vector<int>& houses, vector<int>& heaters) {
        sort(heaters.begin(),heaters.end());
        int ans = 0;
        for(auto house: houses)
        {
            int res = INT_MAX;
            auto pos = lower_bound(heaters.begin(),heaters.end(),house);
            if(pos != heaters.end())
                res = *pos-house;
            if(pos != heaters.begin())
                res = min(res,house-*(pos-1));
            ans = max(ans,res);
        }
        return ans;
    }
};

File: Python/475. Heaters.py
class Solution:
    def findRadius(self, houses, heaters):
        """
        :type houses: List[int]
        :type heaters: List[int]
        :rtype: int
        """
        heaters = sorted(heaters)+[float('inf')]
        ans = i = 0
        for house in sorted(houses):
            while abs(heaters[i+1] - house) <= abs(heaters[i] - house):
                i += 1
            ans = max(ans, abs(heaters[i]-house))
        return ans